### ISSUE TITLE
Allow aggregated tasks within benchmarks

### ISSUE THREAD

**Issue by KennethEnevoldsen** [embeddings-benchmark/mteb#1231]
We currently have only one aggregated task (CQGDupstack), however, we can def. imagien more in the future (e.g. for CoIR in https://github.com/embeddings-benchmark/leaderboard/pull/27).

A proposed solution is to use the benchmark (they are already a group of tasks) and then allow a benchmark to be a `list[task | benchmark]`

This will require updated to the `MTEB.MTEB`, as well as the `create_meta` and potentially for CLI.k

This approach should also solve: https://github.com/embeddings-benchmark/mteb/issues/1171

**Comment by Samoed** [comment]
I think that can be added average result for each subset for multilingual datasets

**Comment by KennethEnevoldsen** [comment]
Not entirely sure what is meant @Samoed ?

**Comment by Samoed** [comment]
Yes, the author of the COIR benchmark wanted an average score for the task. I believe this can be done if all subsets of the task are included in the results. This could also be implemented in the results repository. Currently, there are [some tasks where the average is calculated](https://github.com/embeddings-benchmark/results/blob/11a99065bafa78317967f4490e919727f65a86f1/results.py#L270-L291).

**Comment by KennethEnevoldsen** [comment]
This seems like a quick fix (which I am more than happy to add for now), but it does not specify within benchmark specification within mteb how the scores should be aggregated.

**Comment by isaac-chung** [comment]
> A proposed solution is to use the benchmark (they are already a group of tasks) and then allow a benchmark to be a list[task | benchmark]

It seems like we opted for a different approach, i.e. a new class `AggregateTask`. Wouldn't the original way simplify a bit?

**Comment by KennethEnevoldsen** [comment]
It would have, but then it wouldn't have worked for the leaderboard (with filtering etc.). We can however convert a benchmark quite conveniently to a AggregateTask.
